File: JavaScript/5kyu/PrimeInNumbers.js
function primeFactors(n) {
  var primes = [];

  for (let i = 2; i <= n; i++) {
    let counter = 0;
    while (n % i == 0) {
      counter++;
      n = n / i;
    }
    if (counter >= 1) {
      primes.push(counter > 1 ? `(${i}**${counter})` : `(${i})`);
    }
  }

  return primes.join("");
}

function primeFactors_kata_1(n) {
  for (var i = 2, res = "", f; i <= n; i++) {
    f = 0;
    while (n % i == 0) {
      n /= i;
      f++;
    }
    res += f ? "(" + (f > 1 ? i + "**" + f : i) + ")" : "";
  }

  return res || "(" + n + ")";
}

export { primeFactors };

File: CSharp/5kyu/PrimeInNumbers.cs
using System.Collections.Generic;
using System.Linq;

namespace CSharp._5kyu
{
    public static partial class Kata5
    {
        public static string factors(int lst)
        {
            var result = new Dictionary<int, int>();

            for (int i = 2; i <= lst; i++)
            {
                int counter = 0;
                while (lst % i == 0)
                {
                    counter++;
                    result[i] = counter;
                    lst = lst / i;
                }
            }


            return string.Join("", result.Select(p => string.Format(p.Value > 1 ? "({0}**{1})" : "({0})", p.Key, p.Value)));
        }
    }
}